packages/@aws-cdk/aws-elasticloadbalancingv2/lib/shared/base-load-balancer.ts

```diff
@@ -127,8 +129,29 @@ export abstract class BaseLoadBalancer extends Resource {
 
     const internetFacing = ifUndefined(baseProps.internetFacing, false);
 
+    const hasPrivateSubnets = baseProps.vpc.privateSubnets !== undefined && baseProps.vpc.privateSubnets.length > 0;
+    const hasPublicSubnets = baseProps.vpc.publicSubnets !== undefined && baseProps.vpc.publicSubnets.length > 0;
+    const hasIsolatedSubnets = baseProps.vpc.isolatedSubnets !== undefined && baseProps.vpc.isolatedSubnets.length > 0;
+
+    let vpcSubnetType;
+    if (internetFacing) {
+      if (!hasPublicSubnets) {
+        throw new Error("Internet-facing load balancer requires 'Public' subnets, but none were found.");
+      } else {
+        vpcSubnetType = ec2.SubnetType.PUBLIC;
+      }
+    } else {
+      if (hasPrivateSubnets) {
+        vpcSubnetType = ec2.SubnetType.PRIVATE;
+      } else if (hasIsolatedSubnets) {
+        vpcSubnetType = ec2.SubnetType.ISOLATED;
+      } else {
+        throw new Error("Internal load balancer requires 'Private' or 'Isolated' subnets, but none were found.");
+      }
+    }
+
     const vpcSubnets = ifUndefined(baseProps.vpcSubnets,
-      { subnetType: internetFacing ? ec2.SubnetType.PUBLIC : ec2.SubnetType.PRIVATE });
+      { subnetType: vpcSubnetType });
```

Comment:
I recently updated `Vpc` to have some of this behavior already (private => isolated as a fallback).

Can you try the following and see if that behaves the same?

```ts
const vpcSubnets = ... (internetFacing ? { subnetType: PUBLIC } : {});
```

I.e. leave `subnetType` empty, rely on the `Vpc` default to try private => isolated => public in that order?

Reply:
That just worked, no additional changes to tests needed.